.github/workflows/ci.yml

```diff
@@ -0,0 +1,300 @@
+        run: pacman --noconfirm -Syu
+      # 2. Install the required dependencies
+      - name: Install dependencies
+        run: pacman --noconfirm -S cmake make binutils gcc python3 git vulkan-devel ninja tar unzip which
+      # 3. Checkout the repository, this is done via a github action
+      - name: Checkout repository
+        uses: actions/checkout@master
+      # 4. Prepare nova
+      #    Currently there are 2 different build methods in use with nova,
+      #    so check which one the branch we are on is using and adjust as required
+      - name: Prepare
+        shell: bash
+        run: |
+          set -e
+
+          if [[ -f "build.py" ]]; then
+              # The new build.py method is easy, just run it with python
+              # and we are good to go
+              python3 build.py
+          else
+              # The fiddly part, this will require a manual install of vcpkg... lets do it
+              git clone https://github.com/microsoft/vcpkg vcpkg
+              pushd vcpkg
+              bash bootstrap-vcpkg.sh
+              export PATH=`pwd`:$PATH
+              popd
+              # Now vcpkg should be set up
+              VULKAN_SDK=/usr bash setup-linux.sh
+          fi
+      # 5. Configure with CMake
+      - name: Configure with CMake
+        run: |
+          mkdir build
+          cd build
+          VULKAN_SDK=/usr cmake ..   \
+            -G"Unix Makefiles"       \
+            -DNOVA_TEST=ON           \
+            -DCMAKE_BUILD_TYPE=Debug \
+            -DCMAKE_TOOLCHAIN_FILE=../vcpkg/scripts/buildsystems/vcpkg.cmake
+      # 6. Build with CMake
+      #    We use 3 threads on 2 cores since thats what Github provides as of writing this script,
+      #    for further reference see https://help.github.com/en/articles/virtual-environments-for-github-actions
+      - name: Build with CMake
+        working-directory: build
+        run: cmake --build . -- -j3
+      # 7. Run the unit tests
+      - name: Run unit tests
+        working-directory: build
+        run: ./nova-test-unit --gtest_color=yes
+      # 8. Prepare artifacts for uploading
+      - name: Prepare artifacts
+        run: |
+          # Binary artifacts
+          mkdir artifacts
+          cp LICENSE artifacts
+          cp build/nova-test-unit artifacts
+          cp build/nova-test-end-to-end artifacts
+          cp build/libnova-renderer.a artifacts
+      # 9. Upload binary artifacts
+      - name: Upload binary artifacts
+        uses: actions/upload-artifact@master
+        with:
+          name: binary-linux-gcc
+          path: artifacts
+
+  build-linux-clang:
+    # Even tho we run the real build in an arch linux docker
+    # containerwe need to specify a HostOS
+    runs-on: ubuntu-latest
+
+    # Run the real build on an arch linux docker container
+    # Ubuntu has out of date tools and sometimes system updates
+    # require manual interaction so its unusable for CI
+    container: archlinux/base
+
+    steps:
```

Comment:
Looks like a lot of this is similar to the GCC steps. Can we extract each step into a GitHub action?

Reply:
This is confirmed very similar, however, Github Actions themself are not simple bash scripts but entire javascript steps which do not run in the container specified by the build job, so this is not practical for our use case.

Reply:
You should be able to pull common functionality into a meta-step or meta-job or something so you don't duplicate it.

Reply:
As far as I know there is no such functionality. You can't define step and reuse it. Maybe also because of the fact this is still beta. For now I think this is the best we can get. Feel free to point out if there is a possibility. It could be possible to let the vcpkg build only run once on one system and share that via an artifact, but that would add more steps than it would remove (since system update and such would then be duplicated 3 times).

Reply:
Damn, azure has it but github actions doesn't. I honestly would sooner use a build matrix, as build matrices seem a bit more sane in github actions than in travis.